lib/php-extension/HookAst.cpp

```diff
@@ -0,0 +1,123 @@
+#include "Includes.h"
+
+HashTable *global_ast_to_clean;
+ZEND_API void (*original_ast_process)(zend_ast *ast) = nullptr;
+
+void ast_to_clean_dtor(zval *zv){
```

Comment:
Maybe add comment that this is called to cleanup value of hash table when key is cleaned up? (if I understand it well enough)